Add a TravisCI config

hawkrives opened the pull request:
In my projects, I've found it really helpful to set up a CI (continuous integration) service. As @bauen1 eloquently put it,

> It wouldn't have to actually build stuff, just run the tests which really helps if you commit stuff and are too lazy / don't have the time to do a clean build beforehand.

This PR sets up Travis to start up an OS X 10.12 VM with node 5.1.1 (they have 10.10 through 10.12 available), and then run `npm run build-server`, `npm run build-client`, and `npm run test-local`. I can add the `test-browser` command at a later date.

@felixhageloh: All you will need to do is: 
- open https://travis-ci.org and log in with your GitHub account
- flip the "on" switch next to Uebersicht in the list of repositories

You'll likely want to go into the [Settings](https://travis-ci.org/felixhageloh/uebersicht/settings) screen and turn on the "Build only if .travis.yml is present" option so Travis won't start failing your builds on `master` while you review this PR, as well.

Closes #263.

felixhageloh commented on .travis.yml:
```diff
@@ -0,0 +1,15 @@
+install:
+  - npm install
+
+script:
```
Is it possible to just run `npm test` here?

hawkrives replied:
Yes indeed! 

I broke it apart because I don't have a web browser running on Travis yet. I'll update it to just `npm test`. I'll figure out browsers-on-travis later today, I hope.

(You can see my attempts on my fork, [hawkrives/uebersicht#travis](https://github.com/hawkrives/uebersicht/tree/travis), and on [TravisCI itself](https://travis-ci.org/hawkrives/uebersicht).)

felixhageloh replied:
Cool! I have vague memories of me trying Travis in the past but also not getting browser tests to run. Some pointers might be: I'm using `tape-run` which uses `browser-run` which uses electron - and all on somewhat outdated versions I assume. 

Btw if you just want to run local tests you can just call `npm run test-local`. No build step should be needed

hawkrives replied:
Ah. Thanks.

I think that running the build step would still be useful, though; not as a prerequisite of running the tests, but as a sanity check, just to have the project be _built_ on a remote machine, so that undocumented local dependencies can't creep into the build process.

hawkrives replied:
(I have plans to try and add command-line building of the xcode project, too, but that will almost certainly be a future project, thanks to code signing…)

felixhageloh replied:
cool, I wouldn't worry about it for now - hardy any test for the native part of the app anyway 😳

felixhageloh commented:
awesome thanks :)! had one small question on the first glance

hawkrives commented:
I just rebased to include your master branch's spec fixes, so hopefully this build (https://travis-ci.org/hawkrives/uebersicht/builds/217826657) will pass.

hawkrives commented:
I couldn't get xvfb to run on macOS, _but_ I noticed that run-browser can run in phantomjs instead of electron. 

When I added a `test-browser-phantom` command that runs the browser tests with phantom, [it passed](https://travis-ci.org/hawkrives/uebersicht/builds/219295253), so I've added that to this PR.

So, now that Travis [is running the entire JS test suite](https://travis-ci.org/hawkrives/uebersicht/builds/219316905), I think this is ready for review.

felixhageloh commented:
Awesome thanks!

hawkrives commented:
Woo!

If it ever breaks, feel free to mention me and I'll take a look at it.

hawkrives commented:
@felixhageloh: Don't forget to enable travis on their side, too!

All you should need to do is to

- open https://travis-ci.org and log in with your GitHub account
- flip the "on" switch next to Uebersicht in the list of repositories

I'd also recommend activating the following options in the Settings screen:

- [x] Build only if .travis.yml is present
- [x] Build branch updates
- [ ] Limit concurrent builds (I this this one only applies to the paid service)
- [x] Build pull request updates

felixhageloh commented:
(sorry I have not forgotten about this) Thanks I will look into that this weekend hopefully!
